modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/transport/netty4/Netty4TransportMultiPortIntegrationIT.java

```diff
@@ -64,17 +66,28 @@ protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
         return builder.build();
     }
 
-    public void testThatTransportClientCanConnect() throws Exception {
+    public void testThatTransportClientCanConnect() {
         Settings settings = Settings.builder()
             .put("cluster.name", internalCluster().getClusterName())
             .put(NetworkModule.TRANSPORT_TYPE_KEY, Netty4Plugin.NETTY_TRANSPORT_NAME)
             .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
             .build();
-        // we have to test all the ports that the data node might be bound to
         try (TransportClient transportClient = new MockTransportClient(settings, Arrays.asList(Netty4Plugin.class))) {
-            for (int i = 0; i <= NUMBER_OF_CLIENT_PORTS; i++) {
-                transportClient.addTransportAddress(new TransportAddress(InetAddress.getByName("127.0.0.1"), randomPort + i));
+            final List<TransportAddress> addresses = new ArrayList<>();
+            for (final Transport transport : internalCluster().getInstances(Transport.class)) {
+                for (final TransportAddress transportAddress : transport.boundAddress().boundAddresses()) {
+                    addresses.add(transportAddress);
+                }
+                for (final BoundTransportAddress profileAddresses : transport.profileBoundAddresses().values()) {
+                    for (final TransportAddress transportAddress : profileAddresses.boundAddresses()) {
+                        addresses.add(transportAddress);
+                    }
+                }
             }
+            logger.info("--> all addresses: {}", addresses);
+            final TransportAddress[] targetAddresses = randomNonEmptySubsetOf(addresses).toArray(new TransportAddress[0]);
+            logger.info("--> target addresses: {}", Arrays.toString(targetAddresses));
+            transportClient.addTransportAddresses(targetAddresses);
             ClusterHealthResponse response = transportClient.admin().cluster().prepareHealth().get();
```

Comment:
Is the idea here that prepareHealth broadcasts to all nodes / transport addresses? Are we tricking the TransportClient into thinking that each address is a different node? I see `transportClient.addTransportAddresses` eventually ends up [creating a list of DiscoveryNodes](https://github.com/elastic/elasticsearch/blob/d38e4b028f4a9784bb74de339ac1b877e2dbea6f/server/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java#L188-L199).

Reply:
No, the `TransportClient` ultimately chooses just one address to which to send the request.